### PR TITLE
Eradicating url() calls

### PR DESCRIPTION
**Pull request by arildm**

### CODE REVIEW

**Comment by larowlan** [comment]
looks like dupe of #39?

**Comment by arildm** [comment]
They are related but contain different changes. Didn't see #39 before creating this.

**Comment by larowlan** [comment]
ok, will take a look asap

On 16 October 2014 16:39, Arild notifications@github.com wrote:

> They are related but contain different changes. Didn't see #39
> https://github.com/larowlan/token/pull/39 before creating this.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/larowlan/token/pull/41#issuecomment-59320164.

## 

Lee Rowlands
Senior Drupal Developer
PreviousNext
skype: larowlan
email: larowlan@previousnext.com.au

File: src/Tests/TokenCommentTestCase.php [hunk 1]
```diff
@@ -35,9 +35,9 @@ function testCommentTokens() {
     $parent_comment_path = ltrim($parent_comment_path, '/');
 
     $tokens = array(
-      'url' => url('comment/' . $parent_comment->id(), array('fragment' => 'comment-' . $parent_comment->id(), 'absolute' => TRUE)),
-      'url:absolute' => url('comment/' . $parent_comment->id(), array('fragment' => 'comment-' . $parent_comment->id(), 'absolute' => TRUE)),
-      'url:relative' => url('comment/' . $parent_comment->id(), array('fragment' => 'comment-' . $parent_comment->id(), 'absolute' => FALSE)),
+      'url' => \Drupal::url('entity.comment.canonical', ['comment' => $parent_comment->id()], array('fragment' => 'comment-' . $parent_comment->id(), 'absolute' => TRUE)),
```
Comment (Berdir) [inline src/Tests/TokenCommentTestCase.php]: I think comments have a permalink() method that should return the fragment part?

File: src/Tests/TokenCurrentPageTestCase.php [hunk 2]
```diff
@@ -40,13 +40,13 @@ function testCurrentPageTokens() {
     $node = $this->drupalCreateNode(array('title' => 'Node title', 'path' => array('alias' => 'node-alias')));
     $tokens = array(
       '[current-page:title]' => 'Node title',
-      '[current-page:url]' => url("node/{$node->id()}", array('absolute' => TRUE)),
-      '[current-page:url:absolute]' => url("node/{$node->id()}", array('absolute' => TRUE)),
-      '[current-page:url:relative]' => url("node/{$node->id()}", array('absolute' => FALSE)),
+      '[current-page:url]' => \Drupal::url('entity.node.canonical', ['node' => $node->id()], array('absolute' => TRUE)),
```
Comment (Berdir) [inline src/Tests/TokenCurrentPageTestCase.php]: I suggest you use $node->url('canonical', array('absolute' => TRUE) and so on for this.

File: token.tokens.inc [hunk 3]
```diff
@@ -479,7 +480,7 @@ function token_tokens($type, array $tokens, array $data = array(), array $option
           $replacements[$original] = (int) $count;
           break;
         case 'edit-url':
-          $replacements[$original] = url("admin/structure/types/manage/{$node_type->type}", $url_options);
+          $replacements[$original] = \Drupal::url('entity.node_type.edit_form', ['node_type' => $node_type->type], $url_options);
```
Comment (Berdir) [inline token.tokens.inc]: Also here, $node_type->url('edit-form') should give you the same.

File: token.tokens.inc [hunk 4]
```diff
@@ -808,8 +809,8 @@ function token_tokens($type, array $tokens, array $data = array(), array $option
     foreach ($tokens as $name => $original) {
       switch ($name) {
         case 'url':
-          if (_token_module($type, 'url') == 'token' && $uri = $entity->urlInfo()->toArray()) {
-            $replacements[$original] = url($uri['path'], $uri['options']);
+          if (_token_module($type, 'url') == 'token' && $url = $entity->urlInfo()) {
+            $replacements[$original] = $url;
```
Comment (Berdir) [inline token.tokens.inc:797]: This is not correct, should probably use url() instead, the token replacement needs to be a string.

Comment (Berdir) [inline token.tokens.inc:797]: Same for the nested url tokens below. The current url type expects route name, this passes along url apparently I think type url should expect a Url object, not route_name and so on, like most other API's in 8.x now. So this should pass along array('url' => $entity->urlInfo()) instead and the logic for type url needs to be reworked then.

**Comment by arildm** [comment]
Some tests are failing but we couldn't figure out if they were introduced with these changes or not. We should probably run the tests before these changes, with core branch rolled back to before url() was removed, in order to compare test results.

**Comment by arildm** [comment]
Yay, the last commits pass the tests. Some menu link-related fails remain but they are another issue. Please review, and sorry for the mess!

**Pull request merged**